spec/services/npq/funding_eligibility_spec.rb

```diff
@@ -3,167 +3,417 @@
 require "rails_helper"
 
 RSpec.describe NPQ::FundingEligibility, :with_default_schedules do
-  subject { described_class.new(trn:, npq_course_identifier: npq_course.identifier) }
+  subject { described_class.new(trn:, get_an_identity_id:, npq_course_identifier: course_for_lookup.identifier) }
+
+  let(:course_for_lookup) { npq_course }
+
+  # Having this application in the DB makes sure that other user's applications
+  # don't have an impact on the results. If we were failing to scope our search by
+  # user then this accepted and funded application for the same course would produce
+  # incorrect results
+  let!(:unrelated_application) do
+    create(
+      :npq_application,
+      lead_provider_approval_status: "accepted",
+      eligible_for_funding: true,
+      npq_course: course_for_lookup,
+    )
+  end
+
+  let(:trn) { nil }
+  let(:get_an_identity_id) { nil }
 
   describe "#call" do
-    context "when not previously funded" do
-      let(:trn) { application.teacher_reference_number }
-      let(:application) do
-        create(
-          :npq_application,
-          lead_provider_approval_status: "accepted",
-          eligible_for_funding: false,
-        )
-      end
-      let(:npq_course) { application.npq_course }
+    context "with a trn" do
+      context "when not previously funded" do
+        let(:trn) { application.teacher_reference_number }
+        let(:application) do
+          create(
+            :npq_application,
+            lead_provider_approval_status: "accepted",
+            eligible_for_funding: false,
+          )
+        end
+        let(:npq_course) { application.npq_course }
 
-      it "returns falsey" do
-        expect(subject.call[:previously_funded]).to be_falsey
-        expect(subject.call[:previously_received_targeted_funding_support]).to be_falsey
-      end
-    end
+        it "returns falsey for previously_funded" do
+          expect(subject.call[:previously_funded]).to be_falsey
+        end
 
-    context "when previously funded" do
-      let(:trn) { application.teacher_reference_number }
-      let(:application) do
-        create(
-          :npq_application,
-          eligible_for_funding: true,
-          teacher_reference_number_verified: true,
-        )
-      end
-      let(:npq_course) { application.npq_course }
+        it "returns falsey for previously_received_targeted_funding_support" do
+          expect(subject.call[:previously_received_targeted_funding_support]).to be_falsey
+        end
 
-      before do
-        NPQ::Application::Accept.new(npq_application: application).call
+        context "with a get_an_identity_id" do
+          let(:get_an_identity_id) { SecureRandom.uuid }
+          let(:user) { create(:user, get_an_identity_id:) }
+
+          context "that is not previously funded" do
+            before do
+              create(
```

Comment:
When creating the application here is it linked to the previous one when your checking? basically i wondered if this is having no affect on the call to if its previously funded?

Reply:
Good point, I've updated the spec to make the funded/unfunded specs link much more closely together so that the only thing that differs between the two is the funding status of the application. That way, it has to be that that's making the decision.